src/CloudEventFunctionWrapper.php

```diff
@@ -20,35 +20,230 @@
 use GuzzleHttp\Psr7\Response;
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
+use ReflectionFunctionAbstract;
+use LogicException;
 use RuntimeException;
 
 class CloudEventFunctionWrapper extends FunctionWrapper
 {
+    private const TYPE_LEGACY = 1;
```

Comment:
I'd perfer if the logic for handling legacy events were part of a separate file.

Legacy isn't a type of a HTTP protocol binding for CloudEvents

https://github.com/cloudevents/spec/blob/master/http-protocol-binding.md

Reply:
Added `LegacyEventMapper` to address this. PTAL!

Reply:
The `CloudEventFunctionWrapper` still detects the event type and converts the request to a `CloudEvent`. As we have defined the `FunctionsWrapper` classes to take a `ServerRequestInterface` in `execute`, it makes sense that converting from a Request to a CloudEvent would happen in the wrapper.